aws/data_source_aws_elastic_beanstalk_hosted_zone.go

```diff
@@ -0,0 +1,63 @@
+package aws
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+const unsupportedElasticBeanstalkRegion = "UnsupportedRegion"
+
+// See # http://docs.aws.amazon.com/general/latest/gr/rande.html#elasticbeanstalk_region
+var elasticBeanstalkHostedZoneIds = map[string]string{
+	"us-east-2":      "Z14LCN19Q5QHIC",
+	"us-east-1":      "Z117KPS5GTRQ2G",
+	"us-west-1":      "Z1LQECGX5PH1X",
+	"us-west-2":      "Z38NKT9BP95V3O",
+	"ca-central-1":   "ZJFCZL7SSZB5I",
+	"ap-south-1":     "Z18NTBI3Y7N9TZ",
+	"ap-northeast-2": "Z3JE5OI70TWKCP",
+	"ap-southeast-1": "Z16FZ9L249IFLT",
+	"ap-southeast-2": "Z2PCDNR3VC2G1N",
+	"ap-northeast-1": "Z1R25G3KIG2GBW",
+	"cn-northwest-1": unsupportedElasticBeanstalkRegion,
+	"eu-central-1":   "Z1FRNW7UH4DEZJ",
+	"eu-west-1":      "Z2NYPWQ7DFZAZH",
+	"eu-west-2":      "Z1GKAAAUGATPF1",
+	"eu-west-3":      "Z5WN6GAYWG5OB",
+	"sa-east-1":      "Z10X7K2B4QSOFV",
+}
+
+func dataSourceAwsElasticBeanstalkHostedZone() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceAwsElasticBeanstalkHostedZoneRead,
+
+		Schema: map[string]*schema.Schema{
+			"region": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+		},
+	}
+}
+
+func dataSourceAwsElasticBeanstalkHostedZoneRead(d *schema.ResourceData, meta interface{}) error {
+	region := meta.(*AWSClient).region
+	if v, ok := d.GetOk("region"); ok {
+		region = v.(string)
+	}
+
+	zoneID := elasticBeanstalkHostedZoneIds[region]
```

Comment:
While we appreciate the distinction between unknown and unsupported regions at the current moment this is being developed, I think we should simplify this so we do not need to maintain the difference in the future. Recently we switched the provider so AWS SDK updates with new regions will automatically validate them. We can add manually checking the Regions and Endpoints documentation for this EB Hosted Zone data source to the `CONTRIBUTING.md` New Regions section of our documentation.

```go
zoneID, ok := elasticBeanstalkHostedZoneIds[region]
if !ok {
  return fmt.Errorf("Unsupported region: %s", region)
}
```